src/plugins/data/public/search/aggs/types.ts

```diff
@@ -70,3 +72,25 @@ export interface SearchAggsStart {
   ) => InstanceType<typeof AggConfigs>;
   types: AggTypesRegistryStart;
 }
+
+/** @internal */
+export interface AggExpressionType {
+  type: 'agg_type';
+  value: CreateAggConfigParams;
+}
+
+/** @internal */
+export type AggExpressionFunctionArgs<
+  Name extends keyof AggParamsMapping
+> = AggParamsMapping[Name] & Pick<AggConfigJson, 'id' | 'enabled' | 'schema'>;
```

Comment:
This generic helps to streamline the process of typing arguments for an expression function. It uses the provided `Name` to retrieve the params for the agg from the `AggParamsMapping` below. Then it combines those with `id`, `enabled`, and `schema` from the agg config options -- these arguments will be required for each of the agg type expression functions, however the params will be different between each.